src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final MachineProducer<Truck> truckMachineProducer = new TruckProducer();
+    private final MachineProducer<Excavator> excavatorMachineProducer = new ExcavatorProducer();
+    private final MachineProducer<Bulldozer> bulldozerMachineProducer = new BulldozerProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Truck.class) {
```

Comment:
This approach, using if-else statements to determine the type of class, can lead to scalability issues. If more machine types are introduced in the future, you have to modify this method. Consider using a data structure like Map to associate each class with its producer, helping you to adhere to the Open-Closed Principle.